clients/slo/slo.go

```diff
@@ -0,0 +1,213 @@
+		response, err := c.client.List(ctx, rest.RequestOptions{
+			CustomShouldRetryFunc: rest.RetryIfTooManyRequests,
+			QueryParams:           url.Values{"page-key": {nextPageKey}}})
+		defer closeBody(response)
+		if err != nil {
+			return nil, err
+		}
+
+		var listResponse api.ListResponse
+		nextPageKey, listResponse, err = processListResponse(response, unmarshallFromListResponse)
+		if err != nil {
+			return nil, err
+		}
+
+		retVal = append(retVal, listResponse)
+		haveNextPage = nextPageKey != ""
+	}
+
+	return retVal, nil
+}
+
+// Get gets a complete configuration of SLO with an ID
+func (c Client) Get(ctx context.Context, id string) (api.Response, error) {
+	resp, err := c.client.Get(ctx, id, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	defer closeBody(resp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "get", id, err)
+	}
+	return processResponse(resp, nil)
+}
+
+func (c Client) Create(ctx context.Context, body json.RawMessage) (api.Response, error) {
+	resp, err := c.client.Create(ctx, body, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	defer closeBody(resp)
+	if err != nil {
+		return api.Response{}, err // error message from the client is descriptive enough
+	}
+	return processResponse(resp, nil)
+}
+
+func (c Client) Update(ctx context.Context, id string, body json.RawMessage) (api.Response, error) {
+	getResp, err := c.Get(ctx, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	version, err := getVersion(getResp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	resp, err := c.client.Update(ctx, id, version, body, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	defer closeBody(resp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+	return processResponse(resp, nil)
+}
+
+func getVersion(resp api.Response) (string, error) {
+	var getStr struct {
+		Version string `json:"version"`
+	}
+
+	err := json.Unmarshal(resp.Data, &getStr)
+	if err != nil {
+		return "", err
+	}
+
+	return getStr.Version, nil
+}
+
+// Delete removes configuration for SLO with given ID from a server.
+func (c Client) Delete(ctx context.Context, id string) (api.Response, error) {
+	resp, err := c.client.Delete(ctx, id, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
+	closeBody(resp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	if !rest.IsSuccess(resp) {
+		return api.Response{}, api.NewAPIErrorFromResponse(resp)
+	}
+	return api.NewResponseFromHTTPResponseAndBody(resp, nil), nil
+}
+
+func closeBody(httpResponse *http.Response) {
+	if httpResponse != nil && httpResponse.Body != nil {
+		_ = httpResponse.Body.Close()
+	}
+}
+
+func processResponse(httpResponse *http.Response, transform func([]byte) ([]byte, error)) (api.Response, error) {
```

Comment:
transform is always nil

Reply:
This is part of a separate PR: https://github.com/Dynatrace/dynatrace-configuration-as-code-core/pull/172